C/services/common-plugin-interfaces/python/include/python_plugin_common_interface.h

```diff
@@ -357,11 +357,26 @@ static PLUGIN_INFORMATION *Py2C_PluginInfo(PyObject* pyRetVal)
 		}
 		else if(!strcmp(ckey, "mode"))
 		{
+			// Need to also handle mode values of the form "poll|control"
+			stringstream ss(valStr); 
+			string s;
+
 			info->options = 0;
-			if (!strcmp(valStr, "async"))
+			
+			// Tokenizing w.r.t. pipe '|'
+			while(getline(ss, s, '|'))
 			{
-				info->options |= SP_ASYNC;
+				Logger::getLogger()->debug("%s: mode: Found token %s", __FUNCTION__, s.c_str());
+				if (s.compare("async")==0)
+				{
+					info->options |= SP_ASYNC;
+				}
+				else if (s.compare("control")==0)
+				{
+					info->options |= SP_CONTROL;
+				}
```

Comment:
It would probably be a good idea to have an else branch that logs an unknown option.